lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1069,29 +1076,31 @@ public function remove_class( $class_name ) {
 	 * @return string The processed HTML.
 	 */
 	public function __toString() {
+		/*
+		 * @TODO: What are we intending to measure here? Why check the end of the tag name?
```

Comment:
If the `tag_name_ends_at` is null, it means we are not currently parsing and can short-circuit. This can happen in the following cases:

```php
$p = new WP_HTML_Tag_Processor('<div></div>');
(string) $p;
$p->next_tag('div');
$p->add_class('active');
$p->next_tag('span');
(string) $p;
```

This works because `$this->updated_html` is updated not just in `__toString()` but also in `after_tag()`. If this is confusing while reading the code, let's add this as a comment.

Reply:
I started by documenting this behavior in details, and then I realized that a different check would make the intention crystal clear so I went for that instead:

```php
		// Short-circuit if there are no updates to apply.
		if ( ! count( $this->classname_updates ) && ! count( $this->attribute_updates ) ) {
```

Reply:
this check makes much more sense to me; thank you